src/external_api/exec.rs

```diff
@@ -16,22 +17,20 @@ pub trait ExternalAPI<Input: ToView, Output: ToView> {
 
 // An empty library that implements External Library.
 // This can be used by those controllers that don't rely on a third-party library.
-// Users can define a reconciler as Reconciler<xx, xx, EmptyMsg, EmptyMsg, EmptyLib>.
-pub struct EmptyLib {}
+// Users can define a reconciler as Reconciler<xx, xx, EmptyType, EmptyType, EmptyAPI>.
```

Comment:
What are xx? Could you use the actual type name here?

Reply:
I want to say something like a wildcard. I mean if there's not external api required, the last three generics can be like this, and I don't really care about the first two.